test/routes/src/server/routes/t_launchpad.js

```diff
@@ -944,6 +944,58 @@ describe('The Launchpad API endpoint', () => {
                 done(err);
               });
           });
+
+          context('when snaps are memcached', () => {
+            const urlPrefix = 'https://github.com/anowner/';
+            const repositoryUrl = 'https://github.com/anowner/aname';
+            let snap;
+
+            before(() => {
+              snap = {
+                resource_type_link: `${lpApiBase}/#snap`,
+                self_link: `${lpApiBase}/~test-user/+snap/test-snap`,
+                owner_link: `${lpApiBase}/~test-user`,
+                git_repository_url: repositoryUrl
+              };
+
+              setupInMemoryMemcached();
+              // find snap by url cache will be filled by API call
+
+              // fill snap listing and snapcraft.yaml data caches
+              getMemcached().set(getUrlPrefixCacheId(urlPrefix), [snap]);
+              getMemcached().set(
+                getSnapNameCacheId(repositoryUrl),
+                { name: 'test-snap-name' }
+              );
```

Comment:
These should just do `getMemcached().cache[...] = ...;` rather than getting promises involved.